### PR TITLE
[FIX] sale_purchase: Sales user (no purchasing permissions) cannot cancel their SO

### PR DESCRIPTION
**Pull request by royle-vietnam**
Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by royleviindoo** [comment]
Hi @abh-odoo , can you review this PR, thanks

**Comment by royleviindoo** [comment]
Hi @abh-odoo , can you review this PR, thanks

**Comment by royleviindoo** [comment]
Hi @xavierbol Can you review this PR, thanks

**Comment by xavierbol** [comment]
Hello @royleviindoo ,

Thanks for your work. 🙂 

Can you add a test to reproduce the issue? 🙂

**Comment by royleviindoo** [comment]
> Hello @royleviindoo ,
> 
> Thanks for your work. 🙂
> 
> Can you add a test to reproduce the issue? 🙂

Do you mean automated testing?

**Comment by xavierbol** [comment]
> > Hello @royleviindoo ,
> > Thanks for your work. 🙂
> > Can you add a test to reproduce the issue? 🙂
> 
> Do you mean automated testing?

A unit test in python 🙂 
I did not know the scope but I guess you can check your use case in unit test to be sure the test will no longer be broken in the future. 
The unit test could also help the reviewer to validate your fix. 🙂

**Comment by royleviindoo** [comment]
> > > Hello @royleviindoo ,
> > > Thanks for your work. 🙂
> > > Can you add a test to reproduce the issue? 🙂
> > 
> > 
> > Do you mean automated testing?
> 
> A unit test in python 🙂 I did not know the scope but I guess you can check your use case in unit test to be sure your use case will no longer be broken in the future. The unit test could also help the reviewer to validate your fix. 🙂

@xavierbol please review again, thank you

File: addons/sale_purchase/tests/test_sale_purchase.py [hunk 1]
```diff
@@ -258,3 +259,19 @@ def test_pol_description(self):
...
+        self.env.user.groups_id = [(6, 0, self.env.ref("sales_team.group_sale_salesman").ids)]
+        self.sale_order_1.action_confirm()
+
+        purchase_order = self.env['purchase.order'].sudo().search([('partner_id', '=', self.supplierinfo1.partner_id.id), ('state', '=', 'draft')])
+        purchase_lines = self.env['purchase.order.line'].sudo().search([('sale_line_id', 'in', self.sale_order_1.order_line.ids)])
+
+        self.assertEqual(len(purchase_lines), 1, "Only one purchase line should be created on SO confirmation")
+        self.assertEqual(len(purchase_order), 1, "One purchase order should have been created on SO confirmation")
+
+        wizard = Form(self.env['sale.order.cancel'].with_context({'default_order_id': self.sale_order_1.id})).save()
```
Comment (xavierbol) [inline addons/sale_purchase/tests/test_sale_purchase.py:274]: Thanks 🙂 

We could also check `display_purchase_orders_alert` contains the expected value before calling `action_cancel` method. 🙂 

After that, I will let @odoo/rd-logistics doing the final review. 🙂

Comment (royleviindoo) [reply addons/sale_purchase/tests/test_sale_purchase.py:274]: @xavierbol okey, done

**Comment by amoyaux** [comment]
We should add a group on the field to not display it in the view instead. Anyway if the user doesn't have the right to correct it, he will probably just ignore the warning and another mechanism should exist to warn the purchase representative (I think there is a note atm)

**Comment by royleviindoo** [comment]
> Hello. Thanks for reporting. We should add a group on the field to not display it in the view instead. Anyway if the user doesn't have the right to correct it, he will probably just ignore the warning and another mechanism should exist to warn the purchase representative (I think there is a note atm)

Hi @amoyaux , I have edited it according to your suggestion, please review it for me, thank you

**Comment by royle-viindoo** [comment]
Hi @amoyaux , can you review this PR, thank you

**Comment by royle-viindoo** [comment]
Hi @qdp-odoo , sorry if I bother you. Can you review this PR, thanks.

**Comment by Whenrow** [comment]
robodoo r+
Thank you

**Pull request closed without merging**